### ISSUE TITLE
Table interactions should be driven by operations rather than prose

### ISSUE THREAD

**Issue by gibson042** [tc39/ecma262#2944]
ECMA-262 uses a wide variety of patterns when referencing tabular data—implicitly, by column ordinal position, by [quoted or unquoted] column name, etc. Some examples:
* [ToBigInt](https://tc39.es/ecma262/multipage/abstract-operations.html#sec-tobigint): "Return the value that _prim_ corresponds to in Table 14."
* [CreateIntrinsics](https://tc39.es/ecma262/multipage/executable-code-and-execution-contexts.html#sec-createintrinsics): "Set fields of _realmRec_.[[Intrinsics]] with the values listed in Table 6. The field names are the names listed in column one of the table."
* [OrdinaryFunctionCreate](https://tc39.es/ecma262/multipage/ordinary-and-exotic-objects-behaviours.html#sec-ordinaryfunctioncreate) et al.: "Let _internalSlotsList_ be [_the list-concatenation of « … » and_] the internal slots listed in Table \<number>."
* [Evaluation of `typeof`](https://tc39.es/ecma262/multipage/ecmascript-language-expressions.html#sec-typeof-operator-runtime-semantics-evaluation): "Return a String according to Table 41."
* [ApplyStringOrNumericBinaryOperator](https://tc39.es/ecma262/multipage/ecmascript-language-expressions.html#sec-applystringornumericbinaryoperator): "Let _operation_ be the abstract operation associated with _opText_ and Type(_lnum_) in the following table"
* [UnicodeMatchProperty](https://tc39.es/ecma262/multipage/text-processing.html#sec-runtime-semantics-unicodematchproperty-p): "Assert: _p_ is a Unicode property name or property alias listed in the “Property name and aliases” column of Table 68 or Table 69… Let _c_ be the canonical property name of _p_ as given in the “Canonical property name” column of the corresponding row."
* [TypedArray](https://tc39.es/ecma262/multipage/indexed-collections.html#sec-typedarray): "Let _constructorName_ be the String value of the Constructor Name value specified in Table 71 for this TypedArray constructor."
* [NumericToRawBytes](https://tc39.es/ecma262/multipage/structured-data.html#sec-numerictorawbytes): "Let _n_ be the Element Size value specified in Table 71 for Element Type _type_.… Let _convOp_ be the abstract operation named in the Conversion Operation column in Table 71 for Element Type _type_."

I think it would be better (and probably friendlier to [esmeta](https://github.com/es-meta/esmeta)) to introduce operations for accessing them more consistently. A sketch from https://github.com/tc39/proposal-temporal/pull/2429#discussion_r1004808010 demonstrating use of hypothetical **TableRows**(_table_) and **TableRow**(_table_, _columnNameByWhichToSelect_, _columnValue_) operations:
```patch
-        1. For each row of <emu-xref href="#table-temporal-duration-record-fields"></emu-xref>, except the header row, do
-          1. Append the property name in the Property Name column of the row to _propertyNames_.
+        1. For each Record _field_ of TableRows(<emu-xref href="#table-temporal-duration-record-fields"></emu-xref>), do
+          1. Append _field_.[[Property Name]] to _propertyNames_.
         1. Let _sortedPropertyNames_ be SortStringsByCodeUnit(_propertyNames_).
         1. For each property name _property_ of _sortedPropertyNames_, do
           1. Let _value_ be ? Get(_temporalDurationLike_, _property_).
           1. If _value_ is not *undefined*, then
             1. Set _any_ to *true*.
             1. Set _value_ to ? ToIntegerWithoutRounding(_value_).
-            1. Let _fieldName_ be the field name named in the Field Name column in <emu-xref href="#table-temporal-duration-record-fields"></emu-xref> for Property Name _property_.
+            1. Let _field_ be TableRow(<emu-xref href="#table-temporal-duration-record-fields"></emu-xref>, [[Property Name]], _property_).
+            1. Let _fieldName_ be _field_.[[Field Name]].
             1. Set the field of _result_ whose name is _fieldName_ to _value_.
```

**Comment by syg** [comment]
We discussed this on the editor call. The editors don't feel strongly enough about making the phrasing for referring to specific cells / rows / columns consistent. The existing phrasing is fairly clear.

We also talked about how we'd like 262 to _use_ tables. Our current position is that we like tables when used straightforwardly as lookup tables, which most of the use cases are. We want to discourage iteration of table data. For all the use cases you list here, we plan to specifically improve [CreateIntrinsics](https://tc39.es/ecma262/multipage/executable-code-and-execution-contexts.html#sec-createintrinsics) and [typeof](https://tc39.es/ecma262/multipage/ecmascript-language-expressions.html#sec-typeof-operator-runtime-semantics-evaluation) (via #2915). All other uses seem good and we'd like to leave as-is.
